hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java

```diff
@@ -88,6 +88,8 @@ public Optional<HapiSpecOperation> get() {
         final SupportedContract choice = choices[n % choices.length];
 
         HapiContractCreate op = contractCreate(tentativeContract)
+                .payingWith(key.get())
```

Comment:
Does this mean all payer accounts used with this `RandomContract` provider must be auto-created accounts?

Unclear how this works.

Reply:
Not necessarily, the user of this class can specify the initiator account of the `contract create transaction`, which can be auto-created, normal account, admin account 0.0.2, etc.

Reply:
The point of the tests is to check if auto-created account can sign and pay for such transactions, so i've added this option. Beforehand it was using the default payer and singer (account 0.0.2 of I'm not mistaken)

Reply:
Hmm well if it works, it works! 💯